usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -372,88 +351,91 @@ def fetch_agency_details(agency_id):
-                        + Decimal(item["non_federal_funding_amount"] or 0),
-                        "total_funding_amount": cfdas[cfda_number]["total_funding_amount"]
-                        + Decimal(item["total_funding_amount"] or 0),
+                    clean_cfda_number_str: {
+                        "federal_action_obligation": cfda_dicts[clean_cfda_number_str]["federal_action_obligation"]
+                        + Decimal(transaction["federal_action_obligation"] or 0),
+                        "non_federal_funding_amount": cfda_dicts[clean_cfda_number_str]["non_federal_funding_amount"]
+                        + Decimal(transaction["non_federal_funding_amount"] or 0),
+                        "total_funding_amount": cfda_dicts[clean_cfda_number_str]["total_funding_amount"]
+                        + Decimal(transaction["total_funding_amount"] or 0),
                     }
                 }
             )
         else:
-            cfdas.update(
+            cfda_dicts.update(
                 {
-                    cfda_number: {
-                        "federal_action_obligation": Decimal(item["federal_action_obligation"] or 0),
-                        "non_federal_funding_amount": Decimal(item["non_federal_funding_amount"] or 0),
-                        "total_funding_amount": Decimal(item["total_funding_amount"] or 0),
+                    clean_cfda_number_str: {
+                        "federal_action_obligation": Decimal(transaction["federal_action_obligation"] or 0),
+                        "non_federal_funding_amount": Decimal(transaction["non_federal_funding_amount"] or 0),
+                        "total_funding_amount": Decimal(transaction["total_funding_amount"] or 0),
                     }
                 }
             )
 
-    c = []
-    for cfda_number in cfdas.keys():
+    final_cfda_objects = []
+    for cfda_number in cfda_dicts.keys():
         details = fetch_cfda_details_using_cfda_number(cfda_number)
         if details.get("url") == "None;":
             details.update({"url": None})
-        c.append(
-            {
-                "cfda_number": cfda_number,
-                "federal_action_obligation_amount": cfdas[cfda_number]["federal_action_obligation"],
-                "non_federal_funding_amount": cfdas[cfda_number]["non_federal_funding_amount"],
-                "total_funding_amount": cfdas[cfda_number]["total_funding_amount"],
-                "cfda_title": details.get("program_title"),
-                "cfda_popular_name": details.get("popular_name"),
-                "cfda_objectives": details.get("objectives"),
-                "cfda_federal_agency": details.get("federal_agency"),
-                "cfda_website": details.get("website_address"),
-                "sam_website": details.get("url"),
-                "cfda_obligations": details.get("obligations"),
-            }
+        final_cfda_objects.append(
+            OrderedDict(
```

Comment:
Why impose an order here? Wouldn't this slow performance?

Reply:
The API has been moving in the direction of implementing more of these OrderedDicts to maintain order in responses. I don't believe an OrderedDict will impact performance even for large objects (which this isn't) and it might even be [higher performance](https://mail.python.org/pipermail/python-dev/2017-December/151283.html)

The Python 3.5 dictionary doesn't maintain order, however more recent versions of Python do maintain order and they are more memory efficient compared to the Python 3.5 implementation because they leveraged the internals of OrderedDict.